### PR TITLE
BREAKING CHANGE: change `supportedMethods` to DOMString

### LINKED ISSUE #549: Drop array support for supported methods?
Given what's happening in https://github.com/w3c/browser-payment-api/pull/536, should we consider just dropping support for an array of supportedMethods? And just move to a simple string?

It simplifies the story quite a bit, and the original idea for making it an array was so we could do "supportedMethods: ['visa', 'mc']"?

@domenic was mentioning technically this might not be that bad given how the webidl works...

### PR DESCRIPTION
**Pull request by marcoscaceres**
Closes #549.

Changes `supportedMethods` from a sequence to single DOMString.

***
[Preview](https://s3.amazonaws.com/pr-preview/w3c/browser-payment-api/supportedMethods.html) | [Diff](https://s3.amazonaws.com/pr-preview/w3c/browser-payment-api/0eb169f...2a43dba.html)

### CODE REVIEW

**Comment by zcorpan** [comment]
@marcoscaceres is there a web-platform-tests PR for this change?

**Comment by marcoscaceres** [comment]
Not yet... if it gets accepted, we have to basically change all the tests (just quick find/replace). Waiting on Microsoft to give us the OK.

**Comment by zcorpan** [comment]
I'd particularly like to see a test to assert that the old API is *not* supported (could be in historical.https.html).

**Comment by marcoscaceres** [comment]
Yep, will definitely add those too.

**Comment by marcoscaceres** [comment]
Unrelated to this, but @zcorpan, we should think about have some way getting github to block a pull request unless it has a corresponding test in WPT somehow (unless it's non-normative, of course). Would like to chat more about that at TPAC with you, @plehegar, @sideshowbarker, etc.

**Comment by zcorpan** [comment]
Could maybe have a travis CI check that checks the commit messages (looking for a wpt PR URL) and the PR's labels (blocking for, e.g., "do not merge yet", "waiting for tests").

**Review by domenic — APPROVED** [review]
LGTM! Will let you decide when to merge given concerns about everyone being on-board, and having tests ready.

File: index.html
Comment (zcorpan) [inline index.html]: Missing word "member"

File: index.html
Comment (zcorpan) [inline index.html:1354]: Should this be renamed to `supportedMethod`, now that it only takes a single payment method identifier?

Comment (marcoscaceres) [reply index.html:1354]: I don't know how much breakage we can tolerate here. We are happy for it to break on the Mozilla side, but it's really up to Microsoft, Google, and others relying on Blink as they've shipped.

Comment (zcorpan) [reply index.html:1354]: OK. So `["basic-card"]` would still work as it's casted to a DOMString "basic-card". An array with more than one value would concatenate with a comma, and likely (but not impossibly, as URLs can have commas) end up with an unsupported payment identifier. Is this intentional? Changing the name would be more of a clean slate, but I suppose risks ending up having to support both.

Comment (marcoscaceres) [reply index.html:1354]: > Is this intentional? Changing the name would be more of a clean slate, but I suppose risks ending up having to support both.

Yeah, it's intentional. It's probably as backwards compatible as we are going to get.

Comment (zcorpan) [reply index.html:1354]: You could use a union `(DOMString or sequence<DOMString>)` and throw if the sequence doesn't have exactly 1 item. Then an array with multiple items can't "accidentally" (maybe intentionally, if the payment provider is a bit evil) end up being accepted as a supported payment identifier.

**Review by zkoch — APPROVED** [review]
LGTM

**Comment by mollyd26** [comment]
LGTM

**Comment by mollyd16** [comment]
LGTM

**Comment by marcoscaceres** [comment]
Will write some tests for this over next few days.

File: index.html
Comment (mnoorenberghe) [reply index.html:1354]: > Should this be renamed to supportedMethod, now that it only takes a single payment method identifier?

This already threw me off when just looking at the spec. The plural name with a string type is confusing… I was wondering how we're serializing a list into a string (which would be unusual given we could use an array)… Was an implementer not willing to change it?

**Comment by zcorpan** [comment]
@marcos-travel status on https://github.com/w3c/payment-request/pull/551#issuecomment-309408367 ? AFAICT that aspect is not covered by https://github.com/w3c/web-platform-tests/pull/6512

**Comment by marcoscaceres** [comment]
Thanks for the reminder 😲 I forgot to add it.

**Pull request merged**
